Mage.Sets/src/mage/cards/p/PolygraphOrb.java

```diff
@@ -0,0 +1,138 @@
+import mage.abilities.costs.common.CollectEvidenceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.LookLibraryAndPickControllerEffect;
+import mage.abilities.effects.common.LoseLifeSourceControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.choices.Choice;
+import mage.choices.ChoiceImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.PutCards;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetSacrifice;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class PolygraphOrb extends CardImpl {
+
+    public PolygraphOrb(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{4}{B}");
+
+        // When Polygraph Orb enters the battlefield, look at the top four cards of your library.
+        // Put two of them into your hand and the rest into your graveyard. You lose 2 life.
+        Ability ability = new EntersBattlefieldTriggeredAbility(
+                new LookLibraryAndPickControllerEffect(
+                        4, 2, PutCards.HAND, PutCards.GRAVEYARD));
+        ability.addEffect(new LoseLifeSourceControllerEffect(2));
+        this.addAbility(ability);
+
+        // {2}, {T}, Collect evidence 3: Each opponent loses 3 life unless they discard a card or sacrifice a creature.
+        ability = new SimpleActivatedAbility(new PolygraphOrbEffect(), new GenericManaCost(2));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new CollectEvidenceCost(3));
+        this.addAbility(ability);
+    }
+
+    private PolygraphOrb(final PolygraphOrb card) {
+        super(card);
+    }
+
+    @Override
+    public PolygraphOrb copy() {
+        return new PolygraphOrb(this);
+    }
+}
+
+class PolygraphOrbEffect extends OneShotEffect {
+
+    private static final String DISCARD_CHOICE = "Discard a card";
+    private static final String SACRIFICE_CHOICE = "Sacrifice a creature";
+    private static final String LIFE_LOSS_CHOICE = "Lose 3 life";
+
+    PolygraphOrbEffect() {
+        super(Outcome.Detriment);
+        staticText = "Each opponent loses 3 life unless they discard a card or sacrifice a creature";
+    }
+
+    private PolygraphOrbEffect(final PolygraphOrbEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PolygraphOrbEffect copy() {
+        return new PolygraphOrbEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        for (UUID opponentId : game.getOpponents(controller.getId())) {
```

Comment:
Card rules has additional and interesting info about choices: it must hide discard choice until all players finish their choices:

> While resolving Polygraph Orb’s last ability, your opponent chooses a card to be discarded without revealing it, chooses a creature to be sacrificed, or chooses to do neither. Then that player discards that card, sacrifices that creature, or loses 3 life. Your opponent can always choose to lose 3 life, even if they have cards to discard or creatures to sacrifice.
(2024-02-02)

Another implementation of same ability from Tomb of Annihilation save all choices first:
![IMG_9660](https://github.com/magefree/mage/assets/8344157/f3615c47-a14b-4e85-9d56-2ede6ad2f266)

Another rules example from Plaguecrafter:
> As Plaguecrafter's ability resolves, first the player whose turn it is chooses a creature or planeswalker they control, then each other player in turn order does the same, knowing the choices made before them. Then all the chosen permanents are sacrificed at the same time. Next, each player in the same order who couldn't sacrifice a permanent chooses a card in hand without revealing it, then the chosen cards are discarded at the same time.
(2018-10-05)

So hiding a discard choice until all choices made should have priority over another parts:
* all players make choices from 1/2/3 options;
* real commands execute and result check (call loose life on sacrifice/discard fail);

P.S. Just for info: I think there are another cards with same discard bugs (search “.discard” - if it called inside cycle with another choices then it bugged and must be reworked to make choices first).